### DISCUSSION TITLE
Good way to add a small change to an extension

### DISCUSSION THREAD

**Discussion by sellithy** [neoclide/coc.nvim#5150] [Q&A] [answered]
I describe the "small change" I want to add in another [coc-pyright issue](https://github.com/fannheyward/coc-pyright/issues/1115) but I want to ask in this discussion post about the best way to approach this.
I don't mind implementing this myself. In fact, I'm looking for a good way to add such things in the future. I want to be able to "extend" an already established extension

Options I think work:
1. Pull request the change to the LSP
2. Pull request the change to the Coc extension
3. Create my own extension
4. Create a local patch

I would like to know if someone had to deal with something similar before and what they did

**Comment by fannheyward** [comment, marked as answer]
The best way is added to LS, all editors will benefit.

If the LS refuses to add, you can add to extension, way 2 and 3 is both OK, it's better to PR the change to the origin extension first.
